fe/src/main/java/org/apache/doris/load/DeleteHandler.java

```diff
@@ -0,0 +1,684 @@
+                    // There is no job for delete, set job id to -1
+                    info.add(-1L);
+                    info.add(deleteInfo.getTableId());
+                }
+                info.add(deleteInfo.getTableName());
+                if (!forUser) {
+                    info.add(deleteInfo.getPartitionId());
+                }
+                info.add(deleteInfo.getPartitionName());
+
+                info.add(TimeUtils.longToTimeString(deleteInfo.getCreateTimeMs()));
+                String conds = Joiner.on(", ").join(deleteInfo.getDeleteConditions());
+                info.add(conds);
+
+                if (!forUser) {
+                    info.add(deleteInfo.getPartitionVersion());
+                    info.add(deleteInfo.getPartitionVersionHash());
+                }
+                // for loading state, should not display loading, show deleting instead
+//                if (loadJob.getState() == LoadJob.JobState.LOADING) {
+//                    info.add("DELETING");
+//                } else {
+//                    info.add(loadJob.getState().name());
+//                }
+                info.add("FINISHED");
+                infos.add(info);
+            }
+
+        } finally {
+            readUnlock();
+        }
+
+        // sort by createTimeMs
+        int sortIndex;
+        if (!forUser) {
+            sortIndex = 5;
+        } else {
+            sortIndex = 2;
+        }
+        ListComparator<List<Comparable>> comparator = new ListComparator<List<Comparable>>(sortIndex);
+        Collections.sort(infos, comparator);
+        return infos;
+    }
+
+    public boolean addFinishedReplica(Long transactionId, long tabletId, Replica replica) {
+        writeLock();
+        try {
+            DeleteTask task = idToDeleteTask.get(transactionId);
+            if (task != null) {
+                return task.addFinishedReplica(tabletId, replica);
+            } else {
+                return false;
+            }
+        } finally {
+            writeUnlock();
+        }
+    }
+
+    public void replayDelete(DeleteInfo deleteInfo, Catalog catalog) {
+        Database db = catalog.getDb(deleteInfo.getDbId());
+        db.writeLock();
+        try {
+            writeLock();
+            try {
+                // add to deleteInfos
+                long dbId = deleteInfo.getDbId();
+                List<DeleteInfo> deleteInfos = dbToDeleteInfos.get(dbId);
+                if (deleteInfos == null) {
+                    deleteInfos = Lists.newArrayList();
+                    dbToDeleteInfos.put(dbId, deleteInfos);
+                }
+                deleteInfos.add(deleteInfo);
+            } finally {
+                writeUnlock();
+            }
+        } finally {
+            db.writeUnlock();
+        }
+    }
+
+    // for delete handler, we only persist those delete already finished.
+    @Override
+    public void write(DataOutput out) throws IOException {
+        out.writeInt(dbToDeleteInfos.size());
```

Comment:
Use GSON serde

Reply:
There is problem that using GSON read() have to create a new object, but I wish to keep the old object and only use a readField() method.